src/App.tsx

```diff
@@ -1,61 +1,141 @@
+import React, { useState } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './components/TodoInfo';
+import { User } from './components/UserInfo';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState('0');
+  const [todoList, setTodoList] = useState(todos);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (titleError) {
+      setTitleError(false);
+    }
+
+    if (/^[a-zA-Z0-9 ]*$/.test(event.target.value)) {
+      setTitle(event.target.value);
+    }
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    if (userError) {
+      setUserError(false);
+    }
+
+    setUserId(event.target.value);
+  };
+
+  const addTodoItem = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    let isError = false;
+
+    if (!title) {
+      setTitleError(true);
+      isError = true;
+    }
+
+    if (userId === '0') {
+      setUserError(true);
+      isError = true;
+    }
+
+    if (isError) {
+      return;
+    }
+
+    const id = todos
+      .map(todo => todo.id)
+      .sort((a, b) => a - b)[todos.length - 1] + 1;
```

Comment:
You can make it less:
```suggestion
    const id = todos.sort((prevTodo, nextTodo) => nextTodo.id - prevTodo.id)[0]?.id + 1;
```